agot-bg-game-server/src/common/ingame-game-state/game-data-structure/westeros-card/WinterIsComingWesterosCardType.ts

```diff
@@ -12,16 +12,17 @@ export default class WinterIsComingWesterosCardType extends WesterosCardType {
             westerosDeckI: currentDeckI
         });
 
-        const deck = westerosGameState.game.westerosDecks[currentDeckI];
+        let deck = westerosGameState.game.westerosDecks[currentDeckI];
+        const dontShuffle = westerosGameState.game.revealedWesterosCards - 1;
 
         // Reset discarded state
         deck.forEach(card => { card.discarded = false; });
 
         // Shuffle the deck
-        shuffle(deck);
-
+        deck = deck.slice(0, dontShuffle).concat(shuffle(deck.slice(dontShuffle)));
+        
         // Draw a new card from this deck ...
-        const newDrawnCard = deck.shift() as WesterosCard;
+        const newDrawnCard = deck.splice(dontShuffle, 1)[0] as WesterosCard;
```

Comment:
But this seems to break the normal mode for me. In my test on https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/slice `console.log(animals.slice(-1, 1));` returns an empty array...

Reply:
The command used is actually splice, which functions differently than slice.
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

Doing splice(-1,1) would return the last element in the array instead of an empty one. Still, it's wrong. I fixed it by changing dontShuffle to Math.min(0,westerosGameState.game.revealedWesterosCards - 1). This way nothing gets affected if there are less than 2 revealedWesterosCards, which was the intended behaviour.

Reply:
Random thought: Drawing the last card of the deck (last element in the array) wouldn't actually make a difference because it was only happening during 'Winter is Coming' card. Since the array just got shuffled, any card draw would work. It would only fail if the subsequent draws were also made from the bottom, because the card gets pushed to the end of the array after being selected.

Reply:
> 
> 
> Random thought: Drawing the last card of the deck (last element in the array) wouldn't actually make a difference because it was only happening during 'Winter is Coming' card. Since the array just got shuffled, any card draw would work. It would only fail if the subsequent draws were also made from the bottom, because the card gets pushed to the end of the array after being selected.

Yea thanks for your thoughts. Due to this thoughts I realized how complex WiC becomes. Let's again assume round 9. Here we only have to fix the last card of round 10 and shuffle all others, so we again have to take the "remainingRounds" into account here.